connfig in json, getopt, sqlite driver, elasticsearch driver and much ♥

xaiki opened the pull request:
hey ! THANK YOU, i've been looking for what you built for a long time as everybody seems to have this kind of code and nobody shares.

this patch series tries to standardize the use of getopt for the command line (i think i missed a few still) and puts the configuration in json files (so you don't have to hand edit the .py), it also splits out the db logic so we can have a mysql, sqlite and elasticsearch (incomplete) driver that we specify on the commandline.

(maybe all should go to the config ?)

xaiki committed:
move config to config.py and use it to simplify get_user_ids.py

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
use config.py in streaming.py and use getopt to pass args

xaiki committed:
rename save_to_db to db_mysql and port functions from get_user_ids

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
make get_user_ids use getopts

allow to pass users from file or from command line

xaiki committed:
add sqlite driver

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
add elasticsearch driver

remove dead code from db_mysql

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
Allow following users and hashtags (track)

xaiki committed:
add requirements.txt

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
documentation update in README.md

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
move db files to their own directory

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
port streaming to new DB code,

add signal handler to properly write files on ^C

xaiki committed:
detect connection errors and retry

xaiki committed:
-f is for config file

xaiki committed:
PYNX: normalize users too

xaiki committed:
move arg parsing to config.py and use it in streaming.py

xaiki committed:
ignore .pyc files

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: unify load_* function's naming: use filename for all

xaiki committed:
config: implement load_csv and introduce CSV_FILE config option

xaiki committed:
config: rename user option to users

xaiki committed:
get_user_ids: use new config code

xaiki committed:
port monitoring to new config system

xaiki committed:
port screenshot code to new config API

xaiki committed:
fix sqlite driver

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
allow for empty ids or tracks

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

basilesimon commented:
Hey @xaiki, I am _so_ sorry for leaving this PR open for this long without getting in touch.

Your work looks awesome 🎉 

It's been a while since I used this project and am a little wary of the sheer volume of changes you are proposing.
Have you used the code in this PR for your personal use? I could give it a run, see how I get on.

Thank you for your help

xaiki commented:
Hey Basile !

Don't worry we all have the time we have :)
I have a few more patches in my fork, I've used this a lot a few months ago on a daily basis.

Let me refresh my patches by next week so you don't review code twice.

Cheers

On 26 April 2019 10:22:16 AM GMT-03:00, Basile Simon <notifications@github.com> wrote:
>Hey @xaiki, I am _so_ sorry for leaving this PR open for this long
>without getting in touch.>
>>
>Your work looks awesome 🎉 >
>>
>It's been a while since I used this project and am a little wary of the
>sheer volume of changes you are proposing.>
>Have you used the code in this PR for your personal use? I could give
>it a run, see how I get on.>
>>
>Thank you for your help>
>>
>-- >
>You are receiving this because you were mentioned.>
>Reply to this email directly or view it on GitHub:>
>https://github.com/basilesimon/twitter-tools/pull/1#issuecomment-487054683

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

basilesimon commented:
Super, looking forward to seeing more of your work :)

xaiki committed:
sqlite fix

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
switch config to argparse

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
pep8

xaiki commented:
hey there,
i'm still cleaning up to pass pyflakes and others, but you can have a 
look in:
https://github.com/xaiki/twitter-tools

basically, there are now drivers for different DBs and you could easily 
use many at the same time.

xaiki committed:
port DB to python3

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
db:sqlite: fix import

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
db:sqlite: only create table once

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
db:sqlite: port to python3

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
db:sqlite: sanitize text input by passing to sqlite direct python objects

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: process default config and dbs, add last config term as cmdline arg to make @ladatano happy

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
streaming: fix code and add error handeling

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
2to3

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
2to3

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
add little config tool

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
update README

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: remove debug

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
i never get markdown

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
fix get_user_ids

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
make ids nargs too

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
migrate to logging

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
add tsv driver and make it default

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
allow to pass usernames to commandline

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
get_user_ids is now a bit weird, but will leave it like that

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
support mutiple DB outputs

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
update pynx DB

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: much cleaner and generic driver import

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: better logging, that actually helps

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/generic: cleaner generic method interception

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/generic: now that all goes through multi, really generic is about to disappear

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB: s/save/saveTweet to be more consistent and open the way to save other stuff

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/sqlite: refactor and introduce saveAuthor

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/tsv: actually use filename to write

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/tsv: actual fix

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: dbs actually needed more love to make multi work

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
streaming: fix signal_handler

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
use coloredlogs for more readability

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/multi: separate getattr and fn call so that we properly catch errors

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/sqlite: fix error reporting in execute

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/sqlite: fix saveAuthor upsert syntax

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
streaming: saveAuthor

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: properly implement debug and coloredlogs

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB: generic getAuthor should raise an exception

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB: pynx, split out full text extractor, should probably go to utils

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB: pynx, trivial saveAuthor

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB: sqlite, implement getAuthor and simplify saveAuthor

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB: pynx, make add_tags return tags

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/config: allow for passing instanciated DBs to config

if that happens, don't send in a multi

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
streaming: add debug option

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
get_user_ids: cache ids in SQLITE by default

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: remove debug code

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
get_user_ids: remplace class nonsense with SimpleNamespace

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/generic: back to nothing

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/sqlite: properly handle no results in getAuthor

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: cleaner db loading

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
get_user_ids: bugfix, SimpleNamespace actually needs an import

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
get_user_ids: force sqlite db if the provided one doesn't support author operations

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
rewrite: proper caching, and bulk fetch

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
DB/sqlite: don't reuquire authors to be NOT NULL

so we can store non-existing users

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: row and csv, better logic and logging

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
get_user_ids: actually process csv argument

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
get_user_ids: all lower and tsv output

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
add pydictor/get_user_ids example

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: FetchUsersAction: support for not having a db in config object

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
config: make USERS add elements to 'ids' and add USERS_NOFETCH to bypass queries

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
introduce utils.twitter_login and use it in streaming and get_user_ids

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
implement blocking, supports CSV files

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>

xaiki committed:
Rewrite for testing

we introduce tests, and change database format to acomodate it
this is a huge update but it breaks down somehow like this:
 - introduce tests for db utils and drivers, use pytest, iterate automatically
 - pynx: properly set node attributes and introduce tweets graph (still empty)
 - sqlite: rewrite, more columns, more data, migrations
 - tsv: reorganize
 - DBs: introduce open, close, and _WIPE methods to all (minus mysql)

Signed-off-by: Niv Sardi <xaiki@evilgiggle.com>
